services/src/main/java/org/keycloak/utils/ReservedCharValidator.java

```diff
@@ -22,14 +22,16 @@
 /**
  *
  * @author Stan Silvert
+ * @author Lukas Hanusovsky lhanusov@redhat.com
  */
 public class ReservedCharValidator {
```

Comment:
@ssilvert I changed ReservedChars array and added there 3 new chars. Also Hynek proposed to refactor the whole class and change the for-each loop into regex validation. What do you think?

Reply:
You can code it either way.  If you use regex then you might lose the ability to report which character was disallowed?

Reply:
You can get the disallowed character encountered by the regex via capturing groups. Regex has linear complexity while current `validate` complexity is O(nm).

Reply:
> You can get the disallowed character encountered by the regex via capturing groups. Regex has linear complexity while current `validate` complexity is O(nm).

I doubt that there is actually a meaningful performance benefit because n and m are so small.  And regex solutions often to lead to code that is hard to read.

But like I said, either way is fine with me.